File: pueblo_golf_tournament_web/lib/models/registration.dart
import 'package:json_annotation/json_annotation.dart';

import 'base-model.dart';
part 'registration.g.dart';

@JsonSerializable()
class Registration extends BaseModel {
  final double teamId;
  final double teamCaptainId;
  final double tournamentId;
  final double divisionId;
  final int status;
  final DateTime registrationDate;
  final double? paymentId;

  factory Registration.fromJson(Map<String, dynamic> json) =>
      _$RegistrationFromJson(json);

  Registration(this.teamId, this.teamCaptainId, this.tournamentId,
      this.divisionId, this.status, this.registrationDate, this.paymentId,
      {required super.id});

  Map<String, dynamic> toJson() => _$RegistrationToJson(this);
}

File: pueblo-golf-tournament-api/Entities/Registration.cs
using pueblo_golf_tournament_api.Base;
using pueblo_golf_tournament_api.Utilities.Enums;

namespace pueblo_golf_tournament_api.Entities
{
    public class Registration : BaseEntity
    {
        public long? TeamId { get; set; }
        public long TeamCaptainId { get; set; }
        public long? TournamentId { get; set; }
        public RegistrationStatusEnum Status { get; set; }
        public DateTime RegistrationDate { get; set; }
        public bool IsPayed { get; set; }
        public long? PaymentId { get; set; }
    }
}